sdk/python/feast/feature_store.py

```diff
@@ -318,6 +319,85 @@ def get_historical_features(
+
+        Time travel is based on the configured TTL for each feature view. A shorter TTL will limit the
+        amount of scanning that will be done in order to find feature data for a specific entity key. Setting a short
+        TTL may result in null values being returned.
+
+        Args:
+            entity_df (Union[pd.DataFrame, str]): An entity dataframe is a collection of rows containing all entity
+                columns (e.g., customer_id, driver_id) on which features need to be joined, as well as a event_timestamp
+                column used to ensure point-in-time correctness. Either a Pandas DataFrame can be provided or a string
+                SQL query. The query must be of a format supported by the configured offline store (e.g., BigQuery)
+            feature_refs: A list of features that should be retrieved from the offline store. Feature references are of
+                the format "feature_view:feature", e.g., "customer_fv:daily_transactions".
+            start_date (datetime): Start date for time range of data to filter the feature view before retrieving 
+                features.
+            end_date (datetime): End date for time range of data to filter the feature view before retrieving
+                features.
+
+
+        Returns:
+            RetrievalJob which can be used to materialize the results.
+
+        Examples:
+            Retrieve historical features using a BigQuery SQL entity dataframe
+
+            >>> from feast.feature_store import FeatureStore
+            >>>
+            >>> fs = FeatureStore(config=RepoConfig(provider="gcp"))
+            >>> retrieval_job = fs.get_historical_features(
+            >>>     entity_df="SELECT event_timestamp, order_id, customer_id from gcp_project.my_ds.customer_orders",
+            >>>     feature_refs=["customer:age", "customer:avg_orders_1d", "customer:avg_orders_7d"]
+            >>> )
+            >>> feature_data = retrieval_job.to_df()
+            >>> model.fit(feature_data) # insert your modeling framework here.
+        """
+
+        all_feature_views = self._registry.list_feature_views(project=self.project)
+        try:
+            feature_views = _get_requested_feature_views(
+                feature_refs, all_feature_views
+            )
+        except FeatureViewNotFoundException as e:
+            sys.exit(e)
+
+        provider = self._get_provider()
+        try:
+            warn(
+                "This API is experimental and may change without any deprecation cycle in a future release."
+            )
+            job = provider.get_historical_features_by_view(
+                self.config,
+                feature_views,
+                feature_refs,
+                entity_df,
+                self._registry,
+                self.project,
+                start_date,
+                end_date,
+            )
+        except FeastProviderLoginError as e:
+            sys.exit(e)
```

Comment:
I think we should avoid using `sys.exit(e)` in general. The current implementation of `get_historical_features()` is wrong and will be changed. The exit command will also exit the wrapping process.

Reply:
I'm guessing this is related to #1666